### PR TITLE
ARROW-8745: [C++] Enhance Bitmap::ToString test for big-endian platforms 

### PR DESCRIPTION
**Pull request by kiszk**
This PR fixes the failure of `Bitmap::ToString` that currently takes care of little-endian platforms. This test was added by #6987 

This PR updates the test using an endian-agnostic implementation.

### CODE REVIEW

**Comment by kiszk** [comment]
Or, it would be better to update it as endian independent.

**Comment by kiszk** [comment]
@emkornfield @pitrou  Could you please review this?

**Comment by pitrou** [comment]
WIll do on Monday :-)

**Comment by kiszk** [comment]
Sure, thank you very much

File: cpp/src/arrow/util/bit_util_test.cc [hunk 1]
```diff
@@ -1241,14 +1241,17 @@ TEST(Bitmap, VisitPartialWords) {
 #endif  // ARROW_VALGRIND
 
 TEST(Bitmap, ToString) {
-  uint64_t bitmap_value = 0xCAAC;
-  uint8_t* bitmap = reinterpret_cast<uint8_t*>(&bitmap_value);
-  EXPECT_EQ(Bitmap(bitmap, /*bit_offset*/ 0, /*length=*/34).ToString(),
+  uint64_t bitmap_value = 0xCAAC, bitmap_storage = 0;
```
Comment (pitrou) [inline cpp/src/arrow/util/bit_util_test.cc:1244]: Why not simply:
```c++
uint8_t bitmap[8] = {0xAC, 0xCA, 0, 0, 0, 0, 0, 0};
```
?

**Review by pitrou — CHANGES_REQUESTED** [review]
This looks a bit complicated to me.

File: cpp/src/arrow/util/bit_util_test.cc (hunk 1 above)
Comment (kiszk) [reply cpp/src/arrow/util/bit_util_test.cc:1244]: Oh, absolutely simple and smart!

**Review by pitrou — APPROVED** [review]

**Comment by pitrou** [comment]
Thank you for all the PR's @kiszk !

**Pull request closed without merging**